src/Command/CreateManifestCommand.php

```diff
@@ -0,0 +1,87 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\AppBundle\Command;
+
+use DOMDocument;
+use DOMXPath;
+use Shopware\AppBundle\Service\AnnotationReaderService;
+use Shopware\AppBundle\Service\ManifestCreationService;
+use SimpleXMLElement;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class CreateManifestCommand extends Command
+{
+    private const MANIFEST_DESTINATION = '../development/custom/apps/AppTemplate/manifest.xml';
+
+    protected static $defaultName = 'manifest:create';
+
+    protected static $defaultDescription = 'Command to create the manifest file.';
+
+    private SimpleXMLElement $manifest;
+
+    public function __construct(
+        private ManifestCreationService $manifestCreationService,
+        private AnnotationReaderService $annotationReaderService,
+        private UrlGeneratorInterface $urlGenerator
+    ) {
+        $this->manifest = simplexml_load_file(__DIR__ . '/template.xml');
+
+        parent::__construct(self::$defaultName);
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setName(self::$defaultName)
+            ->setDescription(self::$defaultDescription)
+            ->addOption('secret', 's', InputOption::VALUE_NONE, 'Includes the secret in the manifest.xml.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```

Comment:
I think the `execute` function shouldn't do to much here. The `ManifestCreatorService` should only have a public function `generate` With this the whole command would be:

```
protected function execute(InputInterface $input, OutputInterface $output)
{
    /** @var string $manifest */
    $manifest = $this->manifestCreationService->create():

    // use flysystem or whatever to save the domdocument
    $this->saveToFile($manifest)
}
```

This has the benefit that you don't have to write too many tests for the command but for the service (which should be easier)